### ISSUE TITLE
Meilisearch fails to parse Content-Type header with charset field.

### ISSUE THREAD

**Issue by slightfoot** [meilisearch/MeiliSearch#1816]
**Describe the bug**
Previous version 0.20.0 of Meilisearch would accept content-type `application/json; charset=utf-8` where as new version 0.23.1 only accepts `application/json`. This was problematic for us because our http library would inject the charset in our request.

This has been "worked-around" however our requests are still being sent as utf-8 and we see this as a failure of Meilisearch to correctly decode the content-type header before matching the mime-type.

**Previous Version**
```
{
  "commitSha": "UNKNOWN",
  "buildDate": "2021-03-30T15:13:32.344119665+00:00",
  "pkgVersion": "0.20.0"
}
```

**New Version**
```
{
  "commitSha": "2e2eeb0a424408e27f4223e58afec303c53e1419",
  "commitDate": "2021-10-12T14:47:06+00:00",
  "pkgVersion": "0.23.1"
}
```

**To Reproduce**
1. Make any HTTP request with `Content-Type: application/json; charset=utf-8`

**Expected behavior**
Should compare the mime-type and ignore the charset when limiting content-types in submission.

**MeiliSearch version:**
`0.20.0`

**Comment by slightfoot** [comment]
No. But really this is a bug in your code not correctly parsing the content type header.

**Comment by curquiza** [comment]
This is not a bug but an expected behavior on our side.
See this issue that describes we want to make this behavior consistent everywhere: https://github.com/meilisearch/MeiliSearch/issues/1791

We want the user to be aware of what there are sending as `Content-Type`, that's why we restrict it 🙂 Using one of our SDK will remove this problem for you since they take care of the Content-Type for you.

@gmourier FYI

**Comment by curquiza** [comment]
Fixed in v0.25.0, at the same time we fixed https://github.com/meilisearch/MeiliSearch/issues/1866
